starter-core/src/main/java/io/micronaut/starter/build/gradle/GradleBuild.java

```diff
@@ -16,13 +16,18 @@
 package io.micronaut.starter.build.gradle;
 
 import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.util.CollectionUtils;
+import io.micronaut.starter.build.dependencies.DependencyCoordinate;
+import io.micronaut.starter.build.dependencies.Substitution;
+import io.micronaut.starter.template.RockerWritable;
 import io.micronaut.starter.template.Writable;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-
+import io.micronaut.starter.feature.build.gradle.templates.substitutions;
```

Comment:
what is this?

Reply:
That's the class that rocker generates from the `substitutions.rocker.raw` file